src/node.cc

```diff
@@ -673,11 +673,44 @@ int Init(std::vector<std::string>* argv,
 
 #if !defined(NODE_WITHOUT_NODE_OPTIONS)
   std::string node_options;
+
   if (credentials::SafeGetenv("NODE_OPTIONS", &node_options)) {
-    // [0] is expected to be the program name, fill it in from the real argv
-    // and use 'x' as a placeholder while parsing.
-    std::vector<std::string> env_argv = SplitString("x " + node_options, ' ');
-    env_argv[0] = argv->at(0);
+    std::vector<std::string> env_argv;
+    // [0] is expected to be the program name, fill it in from the real argv.
+    env_argv.push_back(argv->at(0));
+
+    bool is_in_string = false;
```

Comment:
we have `SplitString()`, why don't just fix then use it?

Reply:
I'm not sure that SplitString should contain an actual parsing logic, or double quote/backslash support, does it?

Reply:
For reference this is the other place where SplitString is used - while it wouldn't be broken by using the same logic, in this case it sounds like it would be slightly overkill:

https://github.com/nodejs/node/blob/617f0554beca8f323d7d8922d3f1186a94720dfe/src/node_v8_platform-inl.h#L122

Reply:
But I think it is necessary to abstract a common function like `SplitString` and we should maintain these functions instead of hacking code everywhere

Reply:
If they have the same semantical requirements I'm all for it, but in this case it just doesn't seem the case? I can abstract this behavior in a separate function (`SmartSplitString`), but it'll eventually be used in a single place.

Reply:
@arcanis I think having a common function is probably a good idea.  How about SplitString check's the delimiter specified and does the new logic only if it is a space?  In that way we still have a common function which will do the right thing if you use a space, but otherwise we don't have extra overhead?